src/main/template/vpc.yaml

```diff
@@ -0,0 +1,440 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: This template deploys a VPC. Its IP range is quartered into four parts. The first three parts form private
+  subnets, the fourth is again quartered into three public subnets. Both the private and the public subnets are then
+  spread across three Availability Zones. All the required infrastructure, like internet gateway, nat gateways, and
+  default routes are included. Also, a bastion jump host is deployed in order to make any private machines reachable.
+
+Parameters:
+  cidrBlock:
+    Description: The CIDR range of the VPC. Note that the maximum size of a VPC CIDR range is 16 bits.
+    Type: String
+    Default: 10.0.0.0/16
+
+  firstQuarteringCidrBits:
+    Description: The cidr bits for the four subnets on the first level ("private subnet A", "private subnet B",
+      "private subnet C", "subnet containing public subnets A to C").
+    Type: String
+    Default: 14
+
+  secondQuarteringCidrBits:
+    Description: The cidr bits for the four subnets on the second level ("public subnet A", "public subnet B", "public
+      subnet C", "unused").
+    Type: String
+    Default: 12
+
+  bastionAmi:
+    Description: The AMI used to create the bastion jump host. By default, Amazon Linux 2 is used.
+    Type: AWS::EC2::Image::Id
+    Default: ami-0f3a43fbf2d3899f7
+
+  keyName:
+    Description: The Key Manager key to be added to the list of allowed keys.
+    Type: AWS::EC2::KeyPair::KeyName
+
+  bastionInstanceType:
+    Description: The instance type used for the bastion jump host.
+    Type: String
+    Default: t2.nano
+
+  bastionAuthorizedKeys:
+    Description: A URL pointing to a list of authorized SSH keys to be appended as-is to ~/.ssh/authorized_keys.
+    Type: String
+    MinLength: 1
+
+Resources:
+  vpc:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: !Ref cidrBlock
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} VPC
+
+  internetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Internet Gateway
+
+  vpcGatewayAttachment:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      InternetGatewayId: !Ref internetGateway
+      VpcId: !Ref vpc
+
+  privateSubnetA:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref vpc
+      AvailabilityZone: !Select
+        - 0
+        - Fn::GetAZs: ""
+      CidrBlock: !Select
+        - 0
+        - Fn::Cidr:
+          - !Ref cidrBlock
+          - "4"
+          - !Ref firstQuarteringPartitionBits
```

Comment:
this and other refs still need to be updated. please verify things work by creating a stack.

Reply:
Done. Dammit. Always do these things with search and replace...